### ISSUE TITLE
loss=nan

### ISSUE THREAD

**Issue by AbandonedWarlord** [JiawangBian/sc_depth_pl#39]
Hi,JiawangBian

I constructed my own dataset, and since all the data did not have GT, I only generated pseudo-depth images. I trained the model on my own dataset and I got loss=nan in the first epoch, and it is still the same after reducing the learning rate. I would like to ask for your help to answer this question, I would appreciate it!

**Comment by Vincent630** [comment]
I have same problem,but not using  pseudo-depth images .i trained my own data without any GT by SCDepth_v2, at the early stage, loss reduce normly but it become nan after 80epoch......,if you have any progress please let me konw,iwould appreciate it .

**Comment by Anaaa** [comment]
I manage to stabilize trainings by changing `self.beta` https://github.com/JiawangBian/sc_depth_pl/blob/657dff3ac771edff1e5bf841139d80cc46d38758/models/DepthNet.py#L53 to a value that won't cause numerical overflow (e.g. it could be `self.beta=1/88` or calculate it precisely).

Here https://github.com/JiawangBian/sc_depth_pl/blob/657dff3ac771edff1e5bf841139d80cc46d38758/models/DepthNet.py#L112 if the sigmoid value is almost 0 then `disp = 1/beta`, so `depth = 1/disp = beta`. If a too big value of depth is used here https://github.com/JiawangBian/sc_depth_pl/blob/467935450cb2a6995f2996e1c58392c434339c73/losses/mask_ranking_loss.py#L96 then infinity is hit. The maximal value of float32 is `3.4028235 × 10^38`, for beta being 0.01 the max depth is 100 -> `exp(100)=2,688117142 × 10^43`

**Comment by zhangjy328** [comment]
Hello, have you solved the above problem? I trained on my own dataset, with GT values. Now, the first round of val_loss=0 also occurred. How can I solve this?

**Comment by lvelve** [comment]
'''

    def cal_ranking_loss(self, z_A, z_B, target, call_name=None, epoch=None, batch_idx=None):
        pred_depth = z_A - z_B
        # exp(exp_input) is float32. exp(exp_input) = inf, when exp_input>88. The value of pred_depth should be restricted to the range of [-80, 80].
        pred_depth = torch.clamp(pred_depth, -80.0, 80.0)
        ...
'''
